### ISSUE TITLE
Failure to launch: iOS simulator

### ISSUE THREAD

**Issue by BigAB** [stealjs/steal-cordova#23]
Not sure where the actual root of the problem is, whether it's x-code 9, or the lates ios-sim version or what, but when trying to launch the demo from this reposiroty:

![image](https://user-images.githubusercontent.com/76821/31030289-0a08bad8-a512-11e7-9635-4dd7b1da8bb3.png)

**output**
```bash
** BUILD SUCCEEDED **


No target specified for emulator. Deploying to undefined simulator


Device type "com.apple.CoreSimulator.SimDeviceType.undefined" could not be found.

Done-> cordova emulate ios
Error-> with Parallel tasks1
```

...and no simulator launches.

This is the same error i get when running the **DoneJS** Cordova build (`donejs build cordova`) as well, so currently the guides are not working fully on iOS.

Here is an ios-sim issue possibly related:
https://github.com/phonegap/ios-sim/issues/218

**Comment by marshallswain** [comment]
I'm having this same issue with adding cordova to a donejs app.  

I tried creating a Cordova build separately, using their CLI, and was able to sue the new commands to launch apps.  The new commands are here: https://gist.github.com/marshallswain/e9088589bfca88c43179ad588569cb7d

**Comment by marshallswain** [comment]
The error that I get with a new DoneJS app is this

```
...
...
Done-> cordova plugin add cordova-plugin-transport-security
Success-> with Series tasks
Success-> with Series tasks
Potentially unhandled rejection [1] TypeError: Cannot read property 'dest' of undefined
    at /Users/marshall/Sites/my-project-name/node_modules/steal-cordova/lib/steal-cordova.js:127:25
    at tryCatch (/Users/marshall/Sites/my-project-name/node_modules/es6-promise/dist/es6-promise.js:416:12)
    at invokeCallback (/Users/marshall/Sites/my-project-name/node_modules/es6-promise/dist/es6-promise.js:431:13)
    at publish (/Users/marshall/Sites/my-project-name/node_modules/es6-promise/dist/es6-promise.js:399:7)
    at flush (/Users/marshall/Sites/my-project-name/node_modules/es6-promise/dist/es6-promise.js:126:5)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
Bitovis-MacBook-Pro-2:my-project-name marshall$
```

**Comment by marshallswain** [comment]
I tried `cd my-project/build/cordova` then `cordova run ios --emulator --target="iPad-Air"`, and the build succeeds, but I get the error 

```
** BUILD SUCCEEDED **

No available runtimes could be found for "iPad Air".
```

I think this steal-cordova must be running an older version of Cordova than the latest one I used in with the Cordova CLI.

**Comment by marshallswain** [comment]
Looking further into this error:

```
Potentially unhandled rejection [1] TypeError: Cannot read property 'dest' of undefined
    at /Users/marshall/Sites/my-project-name/node_modules/steal-cordova/lib/steal-cordova.js:127:25
```

stealTools.build is returning undefined

```js
if(buildCordova) {
  buildPromise.then(config => {
    console.log(config) // undefined, undefined
    // stealCordova.build(config)
  })
  // .then(stealCordova.ios.emulate);
}
```

**Comment by marshallswain** [comment]
The problem seems to be here: https://github.com/stealjs/steal-bundler/blob/master/lib/inferred.js#L27

We're looping with `bundles.map`, but since `handler` is `undefined`, that function returns an implicit `undefined`.  Maybe @matthewp or @m-mujica can weigh in on a fix.

**Comment by matthewp** [comment]
I'll take a look, thanks.

**Comment by matthewp** [comment]
@pYr0x Fixed it so that steal-bundler returns the buildResult always... I think maybe I just need to do a release.

**Comment by marshallswain** [comment]
Thanks, @matthewp.  Please let me know.  We need this for a client we're starting tomorrow.

**Comment by matthewp** [comment]
Just released steal-bundler 0.3.3, let me know if that fixes your issue.

**Comment by matthewp** [comment]
Woops, actually the version number is wrong, let me try again.

**Comment by matthewp** [comment]
Ok, 0.3.4 is the number.

**Comment by marshallswain** [comment]
@matthewp that fixed the issue that I was having.  Thanks!

Now I've graduated to having the same issue as @BigAB's first comment.  

This does seem to be a Cordova issue, as the list of iOS devices is empty when I `cd build/cordova` and run `cordova emulate ios --list`.  If I try to `cordova add ios` I get a message stating that it's already added.

I'm able to work around it by opening the project's workspace file in XCode and running the build from there:

![screen shot 2017-10-04 at 7 25 04 am](https://user-images.githubusercontent.com/128857/31181809-72e85e84-a8df-11e7-80dd-b63f30d24dae.jpg)

**Comment by sanylop** [comment]
To solve this problem you only have to go to _platforms/ios/cordova_ and execute this command:
**npm install ios-sim@latest**

**Comment by fayme** [comment]
@sanylop not working

**Comment by Kunj-Choksi** [comment]
@fayme 
any success for this one. Just Tired of launching the app every time from Xcode.
